### PR TITLE
SpeechRecognition fixes

### LINKED ISSUE #2463: [BUG] OfflineSpeechToText Failure: LanguageNotSupported
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Using the unmodified sample provided, I try OfflineSpeechToText and receive a Failure in speech engine - LanguageNotSupported

### Expected Behavior

Expect the sample to work and listen to speech

### Steps To Reproduce

Press the button "StartListenAs" and the error appears in Language Output label

### Link to public reproduction project repository

https://github.com/CommunityToolkit/Maui

### Environment

```markdown
Latest Community toolkit sample 11.0.0, tried on Android 13, 14 and 15.
```

### Anything else?

Italian language devices, language pack downloaded in italian and english.
Tried:
		await speechToText.StartListenAsync(new SpeechToTextOptions
		{
			Culture = new System.Globalization.CultureInfo("en-US"),
			ShouldReportPartialResults = true
		}, CancellationToken.None);

to force "en-US" and the default Culture = CultureInfo.CurrentCulture, that picks "it-IT"

### LINKED ISSUE #2541: [BUG] Polish Language Speech Recognition Not Working
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

I'm unable to get speech to text recognition with polish language, it works perfectly fine offline or online with "en-US". It throws "Request language is not supported error". 
THIS REFERES TO ANDROID AND WINDOWS!

To my dismay, this feature works on Samsung A54 and with their native polish library, however with google package it's unable to work.

This is what i tried in code:
` await _speechToTextService.StartListenAsync(
                new SpeechToTextOptions()
                {
                    // Culture = CultureInfo.GetCultureInfo("PL")
                    // Culture = CultureInfo.CurrentCulture,
                    Culture = CultureInfo.GetCultureInfo("pl-PL"),
                    ShouldReportPartialResults = true,
                },
                CancellationToken.None
            );`

Also I've tried changing current culture of my PC to Polish and it petered away.

These are my windows settings:
Image

Image

I managed to debug external sources and here's the exact line when the error is thrown:
It's 117 line of SpeechRecognizer.cs (scenario with online speech recognition)

Image

### Expected Behavior

Being able to use speech to text with polish language, on windows 10/11 and android.

### Steps To Reproduce

1. Open and run solution. 
2. Press "Start the speech" button.
3. Error should be displayed in a label below.

### Link to public reproduction project repository

https://github.com/wmassalski-arp/Polish-speech-online-recognition-bug/tree/main

### Environment

```markdown
- .NET MAUI CommunityToolkit: 11.1.0
- OS: Windows 10.0.19045 Build 19045, Windows 10 Pro 22H2 
(ANDROID: Android 14, 5.15.104-android13-3-27760517, UP1A.231005.007.A546BXX6BWL3, SM-A546B/DS) 
- .NET MAUI: 9.0
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

 

 ### Linked Issues ###
 

 - Fixes #2541
 - Fixes #2463 

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

 
 

https://github.com/user-attachments/assets/f9481eb7-1633-497b-98c4-ff3c438d4c51

### CODE REVIEW

File: src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/OfflineSpeechToTextImplementation.android.cs [hunk 1]
```diff
@@ -161,4 +178,47 @@ static void SendResults(Bundle? bundle, Action<string> action)
 			action.Invoke(matches[0]);
 		}
 	}
+}
+
+[SupportedOSPlatform("Android33.0")]
```
[the remark this answers is not in the document]
Comment (VladislavAntonyuk) [reply src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/OfflineSpeechToTextImplementation.android.cs:183]: 33 is minimum supported version.

**Comment by ne0rrmatrix** [comment]
I have found a bug with windows implementation of Speech to text page. If you run the app and test speech to text and then close the app it crashes to desktop. Current main does not do this. But current main also does not work for me at all with speech recognition in any language. 


https://github.com/user-attachments/assets/b06e2455-e659-4d80-b556-b348ef05f035

**Review by ne0rrmatrix — COMMENTED** [review]
On windows the changes do exactly what is intended. The offline speech recognition works. the online speech recognition works. Current main I cannot get either offline or online speech recognition to work with windows. 
On android I have zero issues with using this PR. I have been unable to test other languages specifically as I only speak English.

So this is a great fix! I have seen a bug in windows where the app crashes 100 percent of the time if you use online speech recognition. But once that is fixed I think this will be good to go!

**Comment by ne0rrmatrix** [comment]
> This one is primarily the Android-only fix. This issue #1894 will be fixed in another PR

I tested the android version. It is working 100 percent for me with this PR. But I am unable to test other languages because I only speak english. It actually works better on android than on windows. Most likely due to my Mic is not configured properly for silence detection on windows. I have multiple mics on windows and the one I use daily is an xlr mic with a usb interface and I use Nvidia software interface on top of that. So an edge case that also works! If you need anyone to test another language I cannot help. But for what I was able to test everything works On android and windows. If anyone can confirm the language packs working I will be happy to approve this PR.

**Comment by VladislavAntonyuk** [comment]
@ne0rrmatrix I attached video in summary demonstration Ukrainian speech recognition

**Review by ne0rrmatrix — APPROVED** [review]
Looks like another great PR is ready to go! I am happy to approve this PR. I have tested the changes on windows and android. It just works as intended. After looking at the video that @VladislavAntonyuk provided shows other languages I have no concerns.
